app/controllers/users_controller.rb

```diff
@@ -351,10 +351,14 @@ def update
 	end
 
 	def favorite_recipe
-		current_user.recipes.create(:type => "FavoritedRecipe", :meal_type => params[:Type], :title => params[:Title], 
-			:calories => params[:Calories], :time => params[:PrepTime]
-			)
-		current_user.save!
+		previously_favorited = current_user.recipes.where(:type => "FavoritedRecipe", :meal_type => params[:Type], :title => params[:Title], 
+		:calories => params[:Calories], :time => params[:PrepTime])
+		if previously_favorited.blank?
```

Comment:
I’m guessing you’re using the .blank? method to see if there was any previously favorited recipes in the database for the user? Why .blank and not .nil?

Reply:
.blank is used on a collection to check if it is nil or empty (covers both cases)